src/ReflectionFile.php

```diff
@@ -50,6 +50,9 @@ class ReflectionFile
      */
     public function __construct($fileName, $topLevelNodes = null)
     {
+        if (!is_string($fileName)) {
```

Comment:
The purpose of this PR was to address the Scrutinizer issues. This does that **and** it was in the PR from the beginning. 

I'm confused why this is suddenly an issue?

Let me investigate the calling code to see if that is more easily addressed than this. I'll get back to you.